java/src/Ice/src/main/java/com/zeroc/IceSSL/SSLEngine.java

```diff
@@ -624,156 +433,23 @@ else if(properties.getPropertyAsInt("IceSSL.UsePlatformCAs") <= 0)
                     assert(trustManagers != null);
                 }
 
-                //
-                // Create a certificate path validator to build the full
-                // certificate chain of the peer certificate. NOTE: this must
-                // be done before wrapping the trust manager since our wrappers
-                // return an empty list of accepted issuers.
-                //
-                _validator = CertPathValidator.getInstance("PKIX");
-                java.util.Set<TrustAnchor> anchors = new java.util.HashSet<>();
-                for(javax.net.ssl.TrustManager tm : trustManagers)
-                {
-                    X509Certificate[] certs = ((javax.net.ssl.X509TrustManager)tm).getAcceptedIssuers();
-                    for(X509Certificate cert : certs)
-                    {
-                        if(cert.getBasicConstraints() >= 0) // Only add CAs
-                        {
-                            anchors.add(new TrustAnchor(cert, null));
-                        }
-                    }
-                }
-                if(!anchors.isEmpty())
-                {
-                    _validatorParams = new PKIXParameters(anchors);
-                    _validatorParams.setRevocationEnabled(false);
-                }
-
-                //
-                // Wrap each trust manager.
-                //
-                for(int i = 0; i < trustManagers.length; ++i)
-                {
-                    trustManagers[i] = new X509TrustManagerI(this, (javax.net.ssl.X509TrustManager)trustManagers[i]);
-                }
-
                 //
                 // Initialize the SSL context.
                 //
                 _context = javax.net.ssl.SSLContext.getInstance("TLS");
-                _context.init(keyManagers, trustManagers, rand);
+                _context.init(keyManagers, trustManagers, null);
             }
             catch(java.security.GeneralSecurityException ex)
             {
-                throw new PluginInitializationException("IceSSL: unable to initialize context", ex);
+                throw new InitializationException("IceSSL: unable to initialize context", ex);
             }
         }
 
         //
         // Clear cached input streams.
         //
-        _seeds.clear();
         _keystoreStream = null;
         _truststoreStream = null;
-
-        _initialized = true;
-    }
-
-    Certificate[] getVerifiedCertificateChain(Certificate[] chain)
```

Comment:
With the removal of the trust manager wrapper, we can now use the chain returned by the trust manager.